android/app/src/main/java/updated/mysterium/vpn/ui/manual/connect/filter/FilterViewModel.kt

```diff
@@ -0,0 +1,80 @@
+package updated.mysterium.vpn.ui.manual.connect.filter
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import updated.mysterium.vpn.common.extensions.liveDataResult
+import updated.mysterium.vpn.model.filter.NodeFilter
+import updated.mysterium.vpn.model.filter.NodePrice
+import updated.mysterium.vpn.model.filter.NodeQuality
+import updated.mysterium.vpn.model.filter.NodeType
+import updated.mysterium.vpn.model.manual.connect.CountryNodesModel
+import updated.mysterium.vpn.model.manual.connect.PriceLevel
+import updated.mysterium.vpn.model.manual.connect.ProposalModel
+
+class FilterViewModel : ViewModel() {
+
+    private var _proposalsList = MutableLiveData<List<ProposalModel>>()
+    val proposalsList
+        get() = _proposalsList
+
+    private val nodeFilter = NodeFilter()
+    lateinit var nodesModel: CountryNodesModel
+
+    fun applyInitialFilter(countryNodesModel: CountryNodesModel) {
+        nodesModel = countryNodesModel
+        _proposalsList.value = getFilteredProposalList()
+    }
+
+    fun onNodeTypeClicked() = liveDataResult {
```

Comment:
Do we need `liveDataResult` here? I'm sure it will not fail because you just filtering result. Let's just use `liveData {}` structure here and for any similar method.

Reply:
Fix